hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OmUtils.java

```diff
@@ -453,20 +453,32 @@ public static File createOMDir(String dirPath) {
     return dirFile;
   }
 
+  public static RepeatedOmKeyInfo prepareKeyForDelete(OmKeyInfo keyInfo,
+      RepeatedOmKeyInfo repeatedOmKeyInfo) throws IOException {
+    return prepareKeyForDelete(keyInfo, repeatedOmKeyInfo, 0L);
+  }
+
   /**
    * Prepares key info to be moved to deletedTable.
    * 1. It strips GDPR metadata from key info
    * 2. For given object key, if the repeatedOmKeyInfo instance is null, it
    * implies that no entry for the object key exists in deletedTable so we
    * create a new instance to include this key, else we update the existing
    * repeatedOmKeyInfo instance.
+   * 3. Set the updateID to the transactionLogIndex
    * @param keyInfo args supplied by client
    * @param repeatedOmKeyInfo key details from deletedTable
+   * @param trxnLogIndex For Multipart keys, this is the transactionLogIndex
+   *                     of the MultipartUploadAbort request which needs to
+   *                     be set as the updateID of the partKeyInfos.
+   *                     For regular Key deletes, this value should be passed
+   *                     as 0 as the updateID is already set.
    * @return {@link RepeatedOmKeyInfo}
    * @throws IOException if I/O Errors when checking for key
    */
   public static RepeatedOmKeyInfo prepareKeyForDelete(OmKeyInfo keyInfo,
-      RepeatedOmKeyInfo repeatedOmKeyInfo) throws IOException{
+      RepeatedOmKeyInfo repeatedOmKeyInfo, long trxnLogIndex)
```

Comment:
This looks brittle, if someone mistakenly sent for normal key deletes trxnLogIndex we will update it. (I think no harm)
So, make this logic generic for all requests, instead of special cases, and remove the above method.

Reply:
Done.